Headers/Public/VLCTranscoder.h

```diff
@@ -0,0 +1,65 @@
+/*****************************************************************************
+ * VLCTranscoder.h: VLCKit.framework VLCTranscoder implementation
+ *****************************************************************************
+ * Copyright (C) 2018 Carola Nitz
+ * Copyright (C) 2018 VLC authors and VideoLAN
+ * $Id$
+ *
+ * Authors:  Carola Nitz <caro # videolan.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+* Transcoder delegate allows to be notified about transcoding state
+*/
+@protocol VLCTranscoderDelegate <NSObject>
+
+@optional
+/**
+ * Sent by the default notification center whenever the transcoder state has changed.
+ * \param aNotification
+ */
+- (void)transcoderFinishedSucessfully:(BOOL)success;
```

Comment:
The method name should probably be `transcodeFinishedSucessfully:`. The description of the method and documentation description are contradicting itself. The description implies it is called whenever the state has changed, the method name strongly implies it is called when finished successfully but then has a success argument, so maybe it is not finished successfully? Thats very confusing.

Reply:
oh I need to return the instance here of the transcoder object stupid me!